lib/appsignal/minutely.rb

```diff
@@ -1,45 +1,146 @@
+      #   Appsignal::Minutely.probes.register :my_probe, lambda { puts "hello" }
+      #
+      # @example Add a lambda as a probe
+      #   Appsignal::Minutely.probes.register :my_probe, lambda { puts "hello" }
+      #   # "hello" # printed every minute
+      #
+      # @example Add a probe instance
+      #   class MyProbe
+      #     def initialize
+      #       puts "started"
+      #     end
+      #
+      #     def call
+      #       puts "called"
+      #     end
+      #   end
+      #
+      #   Appsignal::Minutely.probes.register :my_probe, MyProbe.new
+      #   # "started" # printed immediately
+      #   # "called" # printed every minute
+      #
+      # @param name [Symbol/String] Name of the probe. Can be used with {[]}.
+      #   This name will be used in errors in the log and allows overwriting of
+      #   probes by registering new ones with the same name.
+      # @param probe [Object] Any object that listens to the `call` method will
+      #   be used as a probe.
+      # @return [void]
+      def register(name, probe)
+        if probes.key?(name)
+          logger.debug "A probe with the name `#{name}` is already " \
+            "registered. Overwriting the entry with the new probe."
+        end
+        probes[name] = probe
+      end
+
+      # @api private
+      def each(&block)
+        probes.each(&block)
+      end
+
+      private
+
+      attr_reader :probes
+
+      def logger
+        Appsignal.logger
+      end
+    end
+
     class << self
-      # List of probes. Probes can be lamdba's or objects that
-      # respond to call.
+      # @see ProbeCollection
+      # @return [ProbeCollection] Returns list of probes.
       def probes
-        @@probes ||= []
+        @@probes ||= ProbeCollection.new
       end
 
+      # @api private
       def start
         stop
         @@thread = Thread.new do
           loop do
             logger = Appsignal.logger
             logger.debug("Gathering minutely metrics with #{probes.count} probes")
-            probes.each do |probe|
+            probes.each do |name, probe|
               begin
-                name = probe.class.name
-                logger.debug("Gathering minutely metrics with #{name} probe")
+                logger.debug("Gathering minutely metrics with '#{name}' probe")
                 probe.call
               rescue => ex
-                logger.error("Error in minutely thread (#{name}): #{ex}")
+                logger.error("Error in minutely probe '#{name}': #{ex}")
               end
             end
             sleep(Appsignal::Minutely.wait_time)
           end
         end
       end
 
+      # @api private
       def stop
         defined?(@@thread) && @@thread.kill
       end
 
+      # @api private
       def wait_time
         60 - Time.now.sec
       end
 
-      def add_gc_probe
-        probes << GCProbe.new
+      # @api private
+      def register_garbage_collection_probe
```

Comment:
Would it maybe make sense to move the `GCProbe` to it's own file?  Somewhere in `lib/appsignal/probes` or something like that?

Reply:
We could create a new hook for it?

Reply:
Ah yeah might be good enough for now indeed. Will you do that? Or shall I take care of it?